libs/category/guides/architecture/filtering.md

```diff
@@ -0,0 +1,27 @@
+# Filter Architecture
+
+## Use Cases
+
+As a user, I would like to be able to apply filters to a collection of items of a category.
+As a user, I would like to know what filters are currently applied to the category. 
+As a user, I would like to see the filters applied IMMEDIATELY, without waiting for a [response from a backend platform (input-delay).](https://web.dev/fid/)
+
+## Features
+- [Filter Types](./filtering/types.md)
+  - [Equal](./filtering/types/equal.md)
+  - [Range](./filtering/types/range.md)
+- [Idempotence](./filtering/idempotence.md)
+
+## Implementation Details
+
+When an action is dispatched that would modify the "applied" filter state of the application, we intentionally optimistically apply the filters to the state of the category page. This means the following occurs in following order:
+
+1. An action modifying the applied filter state is dispatched
+2. Reducers computing the next state ASSUMING that the associated filters will be applied/removed.
```

Comment:
"Reducers compute the..."

Reply:
I think that "the next state" is too ambiguous. It should say that it adds the applied filter from the action to the applied filters in state. Even though "ASSUMING that the associated filters will be applied/removed" implies that the applied filter from the action will be added to state, I think that should be an additional point to explain why the action is added to state instead of expecting the reader to draw that conclusion for themselves. Something like: "Reducers add/remove the applied filter from the action to state. This assumes that the associated filters will be applied/removed successfully on the backend."

Reply:
8b2f50a1